### ISSUE TITLE
Mirror padding for layers.Padlayer

### ISSUE THREAD

**Issue by zzd1992** [Lasagne/Lasagne#766]
layers.Padlayer currently just support zero padding. I think it is necessary to add mirror padding mode to layers.Padlayer.
    Mirror padding is often used in image processing, especially in low level image processing. For example, I am training CNN for image super resolution task. I think zero padding will cause some border patterns which would hurt the performance. In such a case, mirror padding may be better than zero padding.

**Comment by f0k** [comment]
Good call. It's also used in http://openreview.net/forum?id=rJiNwv9gg, for example. We'd appreciate a pull request!

**Comment by alexjc** [comment]
I've been using @ajbrock's implementation:
https://gist.github.com/ajbrock/a3858c26282d9731191901b397b3ce9f

It's slower you're right, it's sometimes worth the cost!

**Comment by zzd1992** [comment]
I have implemented it in a similar may by myself.
Bu t still thank you for your help. @alexjc

**Comment by f0k** [comment]
Great to hear you solved it. We'd still welcome a PR extending `PadLayer` to support mirror padding :)

**Comment by ajbrock** [comment]
^ I'll see about putting that together sometime this month <_<
